prusti-contracts/prusti-specs/src/extern_spec_rewriter/impls.rs

```diff
@@ -340,16 +329,28 @@ mod tests {
         }
 
         #[test]
-        fn generics_not_supported() {
+        fn generics_supported() {
             let mut inp_impl: syn::ItemImpl = parse_quote!(
                 impl MyTrait<I> for MyStruct {
                     fn foo(&mut self, arg1: I);
                 }
             );
 
-            let rewritten = rewrite_extern_spec_internal(&mut inp_impl);
+            let rewritten = rewrite_extern_spec_internal(&mut inp_impl).unwrap();
+
+            let newtype_ident = &rewritten.generated_struct.ident;
+            let expected_impl: syn::ItemImpl = parse_quote! {
+                impl #newtype_ident <> {
+                    #[prusti::extern_spec = "trait_impl"]
+                    #[allow(unused, dead_code)]
+                    #[prusti::trusted]
+                    fn foo(_self: &mut MyStruct, arg1: I) {
+                        <MyStruct as MyTrait<I>> :: foo :: <>(_self, arg1)
```

Comment:
Where is the type parameter `I` declared after rewriting? Is it somehow part of `newtype_ident`?

Reply:
I did a double take here too originally—`I` is ([was](https://github.com/juliand665/prusti-dev/commit/76e6f6598f6161d0afa5ce23ce0fa03ebfe90a73)) in fact not a type parameter! Hence why it's not where you'd expect. I think it makes sense to test both a blanket `impl<I>` and a specific `impl` though. I'll update the tests accordingly, and rename `I` in the non-generic case to not break conventions ;)